lib/server.js

```diff
@@ -117,6 +117,7 @@ function Server(options) {
 
     this.onceNext = !!options.onceNext;
     this.strictNext = !!options.strictNext;
+    this.earliestChain = [];
```

Comment:
Chain is really expensive, I'll post numbers below.

Reply:
Yes, even more so now though. We've discovered that (unsurprisingly in retrospect) breaking the event loop between the load shedding evaluation and inflight request accounting introduces a race condition. Depending on how the accounting done, this will result in either too many requests being shed or not enough.

Forcing them to be synchronous appears to be effective at maintaining the correct inflight request count under load.